amy/trainings/views.py

```diff
@@ -74,16 +79,73 @@ def get_context_data(self, **kwargs):
         context["form"].helper = context["form"].create_helper
         return context
 
+    def form_valid(self, form):
+        person = form.cleaned_data["trainee"]
+        event = form.cleaned_data["event"]
+        result = super().form_valid(form)
+        try:
+            run_instructor_training_completed_not_badged_strategy(
+                instructor_training_completed_not_badged_strategy(person),
+                request=self.request,
+                person=person,
+                training_completed_date=event.end if event else None,
+            )
+        except EmailStrategyException as exc:
+            messages.error(
+                self.request,
+                f"Error when running instructor training completed strategy. {exc}",
+            )
+        return result
+
 
 class TrainingProgressUpdate(RedirectSupportMixin, OnlyForAdminsMixin, AMYUpdateView):
     model = TrainingProgress
     form_class = TrainingProgressForm
     template_name = "trainings/trainingprogress_form.html"
 
+    def form_valid(self, form):
+        person = form.cleaned_data["trainee"]
+        event = form.cleaned_data["event"]
+        try:
+            run_instructor_training_completed_not_badged_strategy(
+                instructor_training_completed_not_badged_strategy(person),
+                request=self.request,
+                person=person,
+                training_completed_date=event.end if event else None,
+            )
+        except EmailStrategyException as exc:
+            messages.error(
+                self.request,
+                f"Error when running instructor training completed strategy. {exc}",
```

Comment:
I'm not sure this message will make much sense to the user, so I suggest something more straightforward.
```suggestion
                f"Error when creating or updating scheduled email. {exc}",
```

Reply:
Good suggestion, thank you @elichad!